### PR TITLE
Feature/v75 jf updates

### PR DESCRIPTION
**Pull request by cookersjs**
- Shared groovy library usage
- Generates FINAL graph database and stores in downloads folder
- Outputs file counts for diagram JSON files

### CODE REVIEW

**Review by SolomonShorser-OICR — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'mysqlPass', usernameVariable: 'mysqlUser')]){
+						withCredentials([usernamePassword(credentialsId: 'neo4jUsernamePassword', passwordVariable: 'neo4jPass', usernameVariable: 'neo4jUser')]){
+				 			sh "java -jar target/diagram-converter-jar-with-dependencies.jar --graph_user $neo4jUser --graph_password $neo4jPass --rel_user $mysqlUser --rel_password $mysqlPass --rel_database ${env.REACTOME_DB} --output ./${env.OUTPUT_FOLDER}"
+					        	// Create archive that will be stored on S3.
+							sh "tar -zcf diagrams-v${releaseVersion}.tgz ${env.OUTPUT_FOLDER}/"
+							// Restart tomcat7 and neo4j services after updates were made to graph db.
+					        	sh "sudo service tomcat7 stop"
+							sh "sudo service neo4j stop"
+							sh "sudo service neo4j start"
+							sh "sudo service tomcat7 start"
```
Comment (jweiser) [inline Jenkinsfile:49]: I think you can do these as "restart" rather than "stop" and "start"?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+				// Download previous diagram-converter output files and extract them.
+				sh "aws s3 --no-progress cp s3://reactome/private/releases/${previousReleaseVersion}/diagram_converter/${previousDiagramsArchive} ${previousReleaseVersion}/"
+				dir("${previousReleaseVersion}"){
+					sh "tar -xf ${previousDiagramsArchive}"
+				}
+				// Output number of JSON diagram files between releases.
+				def currentDiagramsFileCount = findFiles(glob: "${env.OUTPUT_FOLDER}/*").size()
+				def previousDiagramsFileCount = findFiles(glob: "${previousReleaseVersion}/${env.OUTPUT_FOLDER}/*").size()
+				echo("Total diagram files for v${releaseVersion}: ${currentDiagramsFileCount}")
+				echo("Total diagram files for v${previousReleaseVersion}: ${previousDiagramsFileCount}")
```
Comment (jweiser) [inline Jenkinsfile:73]: Can/should there be an automatic check to make sure the file counts are equal or if there's a change it's expected?

File: Jenkinsfile (hunk 1 above)
Comment (cookersjs) [reply Jenkinsfile:49]: Yea I think I will add that to the `visudo` commands allowed for Jenkins. Originally I thought that keeping it as discrete start/stops would be better, but there isn't much of a reason for that

File: Jenkinsfile (hunk 2 above)
Comment (cookersjs) [reply Jenkinsfile:73]: I think thats a good idea in the future, but really this is just supposed to provide a shortcut for evaluating that the step ran correctly - normally for this step you just confirm the file counts are _around_ the same (if not slightly increased) between releases. I know for the Perl version of things that we had a 10% threshold for things like orthopairs, but we also had issues with automating that check.

Comment (jweiser) [reply Jenkinsfile:73]: Okay, just a thought.

**Pull request merged**